### DISCUSSION TITLE
Using SQL Api

### DISCUSSION THREAD

**Discussion by sottey** [TriliumNext/Trilium#4239] [Q&A] [answered]
I apologize if I have missed this documentation, but I have looked for examples or docs for calling the DB directly from the backend API and have been unable to figure it out.

Does anyone have something they can point me to that just shows a simple execute call?

Thanks!
Sean

**Comment by sottey** [comment, marked as answer]
Never fails. The second I post I then figure it out. Posting for future me when I forget:

```
async function dbSelect(query) {
        await api.runOnBackend((query) => {
            api.log(query);
            const results = api.sql.execute(query);
            api.log(results);
         }, [query]);
}

await dbSelect("select * from options");
```

**Comment by zadam** [comment]
`execute()` is meant to be used for INSERT, UPDATE etc.

To fetch data, I'd recommend (in the api.sql):

```js
function getRows(query, params = []) {...}
function getRow(query, params = []) {...}
function getRowOrNull(query, params = []) {...}
function getValue(query, params = []) {...}
```
